marlowe-playground-client/src/Halogen/Blockly.purs

```diff
@@ -186,6 +206,31 @@ handleAction GetCode = do
   where
   unexpected s = "An unexpected error has occurred, please raise a support issue at https://github.com/input-output-hk/plutus/issues/new: " <> s
 
+handleAction BlocklyEvent = raise CodeChange
+
+blocklyChanges :: forall m. MonadAff m => Workspace -> EventSource m Action
+blocklyChanges workspace =
+  EventSource.effectEventSource \emitter -> do
+    listener <-
+      eventListener \event ->
+        let
+          isCreateEvent = isJust $ CreateEvent.fromEvent $ spy "BlocklyEvent" event
+
+          isMoveEvent = isJust $ MoveEvent.fromEvent event
+
+          isChangeEvent = isJust $ ChangeEvent.fromEvent event
+        -- Blockly can fire all of the following events https://developers.google.com/blockly/guides/configure/web/events
+        -- but we only care for the ones that may change the contract.
+        -- TODO: the move event by its own shouldn't necesarly cause a change in the contract
+        -- we should check if the string property "newParentId" is defined
+        in
+          if isCreateEvent || isMoveEvent || isChangeEvent then
```

Comment:
```purescript
do
  mEvent = CreateEvent.fromEvent event <|> MoveEvent.fromEvent event <|> ChangeEvent.fromEvent event
  for_ mEvent \event -> EventSource.emit emitter BlocklyEvent
```

Reply:
Can't use Alt because the events type differ, after discussing in call we decided to move isCreateEvent to the CreateEvent module